### PR TITLE
Propose internal and exported data model, hyperlink solution, spec for curation tool

### PR DESCRIPTION
**Pull request by dhh1128**
This PR is a response to #43 and to a [thoughtful Google doc](https://docs.google.com/document/d/1QQ6376vFB3g8ScGHB2ONwAqSFc1UdC0_nfnyHzTs0Nk/edit) from @RieksJ. It attempts to address several next steps proposed by @vinomaster. Specifically:

1. It proposes a detailed and concrete data model for each phase of the data lifecycle. The [ingestible data model](https://github.com/dhh1128/ctwg/blob/master/docs/ingestible-data-model.md) is similar to the previously agreed-upon issue template for adding new terms, with only a few refinements. The [internal data model](https://github.com/dhh1128/ctwg/blob/master/docs/internal-data-model.md) is new. It shows how normalized data will work, how terms and concepts relate to each other, and how they can be integrated with a [comprehensive hyperlink strategy](https://github.com/dhh1128/ctwg/blob/master/docs/hyperlinks.md). The [exported data model](https://github.com/dhh1128/ctwg/blob/master/docs/exported-data-model.md) describes how data can be emitted from internal tools in a format that standard publication tools like MkDocs, SpecUp, or custom publication tools like Docusarus, can transform the content into polished artifacts. It also explains what [extra metadata will be available after export](https://github.com/dhh1128/ctwg/blob/master/docs/exported-data-model.md#generated-fields), and where it comes from. This includes a resolution to the hovertext question, as well as to Rieks's desire for human-friendly versioning.

2. It proposes a [comprehensive hyperlink strategy](https://github.com/dhh1128/ctwg/blob/master/docs/hyperlinks.md), as noted above. This strategy explains the format of hyperlinks of various types, at each stage in the data lifecycle.

3. It contains a [rough spec for the internal tooling](https://github.com/dhh1128/ctwg/blob/master/toipterm/spec.md) that will facilitate key behind-the-scenes workflows, and that will enforce business rules we care about. This is an important evolution in our thinking; up till now, almost all of our energy has been spent on the late stages of the lifecycle, where tools like MkDocs, SpecUp, and Docusaurus will transform data into publishable artifacts. I have felt like publication is the easier part of the problem, and that internal curation is more difficult and unspecified. So the spec is an attempt to plug that gap.

4. It explains some aspects of the [curation process](https://github.com/dhh1128/ctwg/blob/master/docs/curation.md) and the duties of [curators](https://github.com/dhh1128/ctwg/blob/master/docs/curators.md).

5. It updates the [issue templates](https://github.com/dhh1128/ctwg/issues/new/choose). It also simplifies README.md, adding direct links that will start common workflows instead of just describing those workflows in the abstract.

6. It describes [tagging conventions](https://github.com/dhh1128/ctwg/blob/master/docs/hash-tags.md), linking to the officially approved TOIP tags and explaining how and why we extend that set.

7. It describes [how status of individual terms and concepts is managed](https://github.com/dhh1128/ctwg/blob/master/docs/term-record.md#status).

8. It describes how we will accomplish the v2 feature of having multiple glossaries in a single scope, and multiple scopes in the overall corpus.

9. It retains Rieks's concept of arbitrary concept types, although it doesn't elaborate it. (It leaves a place for it in the design.)

10. It is in harmony with Dan's ask that we stay more tightly aligned with MkDocs and SpecUp. However, it does this by making the distinction between glossary publication and data export more crisp; MkDocs and SpecUp are publication tools that transform markdown, whereas the internal terminology tooling is more like a database that emits markdown that publication tools can consume. This means that the CTWG has a looser relationship to publication in general, which in turn means that if someone wants to publish data with additional tools like Docusaurus, they can easily do so.

I believe that if we accepted this PR, we'd be taking a number of next steps that Dan outlined in #43, and that it would represent substantial forward progress. However, it is undoubtedly imperfect. For example, it doesn't explicitly address the internal dissonance we are having about whether our system should be centralized or decentralized in terms of where curation happens. Because this PR has so many moving parts, I suggest that we discuss interactively in a CTWG meeting.

### CODE REVIEW

File: docs/curation.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+* Tags are normalized
+
+### Group Curation
+
+Beyond simple rules enforced by tooling, some issues that your group might examine during its own curation include things like:
+
+* Are we choosing a word or phrase that has a different meaning to others -- and that therefore would be prone to misunderstanding?
+* Do we have a crisp, terse definition?
+* Have we identified the defining characteristics of a concept, such that we can say with confidence whether X is or is not an example?
+* Have we provided usage exmaples?
```
Comment (talltree) [inline docs/curation.md]:
```suggestion
* Have we provided usage examples?
```

File: docs/exported-data-model.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+The following fields are automatically added to each exported file by the export algorithm:
+
+* Tags U: The union of the Tags field and the Concept Tags field.
+* Tags I: the intersection of the Tags field and the Concept Tags field.
+* Hovertext: The first sentence of the Definition field.
+* Contributors: A list of all github handles that have submitted content for this record.
+* Synonyms: A list of [local hyperlinks](hyperlinks.md#local-links) to other terms that share the same definition.
+* Term commit: The latest commit hash of the term record.
+* Term version: How many unique versions of the term record exists.
+* Concept commet: The latest commit hash of the concept record.
```
Comment (talltree) [inline docs/exported-data-model.md]:
```suggestion
* Concept commit: The latest commit hash of the concept record.
```

File: docs/term-conventions.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+### Acronyms
+
+Acronyms should be defined in the same glossary entry as their expanded form. They should be written in all caps, without periods between them. This means we would expect a single entry in the glossary to contain both "Trust Over IP" and "TOIP" -- and we would not expect to find "T.O.I.P" or "ToIP" anywhere.
+
+### Apostrophes
+
+Contrary to popular mistakes, plurals are not preceded by an apostrophe in English: "three bikes" -- NOT "three bike's"; "many NFTs (non-fungible tokens) are traded in specialized exchanges" -- NOT "many NFT's..."
+
+In English, nouns are converted into possessive form with an apostrophe: "my bike's tire." However, pronouns (he, she, it, we) are NOT: his, hers, its, ours. (Incorrectly inserting an apostrophe in the possessive form of "it" is perhaps the most common misspelling in the English language.
```
Comment (talltree) [inline docs/term-conventions.md]:
```suggestion
In English, nouns are converted into possessive form with an apostrophe: "my bike's tire." However, pronouns (he, she, it, we) are NOT: his, hers, its, ours. (Incorrectly inserting an apostrophe in the possessive form of "it" is perhaps the most common misspelling in the English language.)
```

**Comment by dhh1128** [comment]
Approved on CTWG call, 12 Apr 2021

**Pull request merged**